lib/db/leveldb_dbinstance.go

```diff
@@ -217,19 +219,19 @@ func (db *Instance) updateFiles(folder, device []byte, fs []protocol.FileInfo, l
 		if err != nil {
 			panic(err)
 		}
-		// The Invalid flag might change without the version being bumped.
-		if !ef.Version.Equal(f.Version) || ef.Invalid != f.Invalid {
-			if isLocalDevice {
-				localSize.removeFile(ef)
-				localSize.addFile(f)
-			}
 
-			t.insertFile(folder, device, f)
-			if f.IsInvalid() {
-				t.removeFromGlobal(folder, device, name, globalSize)
-			} else {
-				t.updateGlobal(folder, device, f, globalSize)
-			}
+		// update the DB entry if the file has changed
```

Comment:
Yeah I think the "if" is vestigial so this should be fine. The comment above is incorrect though, there is no "if the file has changed" condition here, other than maybe the fact that we're here at all. Remove, I think,